src/utils/AuthProvider.tsx

```diff
@@ -103,6 +103,25 @@ export default function AuthProvider({
         options,
       }); // will trigger onAuthStateChange to update the session
 
+      // This code is largely taken from https://github.com/orgs/supabase/discussions/1282#discussioncomment-5230475
+
+      // User exists, but is fake. See https://supabase.com/docs/reference/javascript/auth-signup
+      if (
+        value.data.user &&
+        value.data.user.identities &&
+        value.data.user.identities.length === 0
+      ) {
```

Comment:
we can replace this somewhat hacky solution with a call to `check_email_exists` (which now works after changing the security level to Definer), which could make it cleaner and easier for future dev team to refactor if they decide to change or add login methods